src/main/scala/nodes/learning/GaussianMixtureModel.scala

```diff
@@ -16,11 +17,16 @@ import workflow.{Transformer, Estimator}
  * @param variances Cluster variances (diagonal)
  * @param weights Cluster weights.
  */
-class GaussianMixtureModel(
-    val means: DenseMatrix[Double],
-    val variances: DenseMatrix[Double],
-    val weights: DenseVector[Double])
-  extends Transformer[DenseVector[Double],DenseVector[Double]] with Logging {
+case class GaussianMixtureModel(
+    means: DenseMatrix[Double],
+    variances: DenseMatrix[Double],
+    weights: DenseVector[Double],
+    weightThreshold: Double = 1e-4)
+  extends Transformer[DenseVector[Double],DenseVector[Double]] {
+
+  private val gmmMeans = means.t
```

Comment:
hmm good question. I think when I initially wrote this the matlab code I was translating and the native gmm code expected the transpose of each other as inputs, and we wanted them to work the same given the same input.